Update apiDefinition.swagger.json with First 5 actions

jokano12 opened the pull request:
---
### When submitting a connector, please make sure that you follow the requirements below, otherwise your PR might be rejected. We want to make you have a well-built connector, a smooth certification experience, and your users are happy :) 

- [ ] I attest that the connector works and I verified by deploying and testing all the operations. 
- [ ] I attest that I have added detailed descriptions for all operations and parameters in the swagger file.
- [ ] I attest that I have added response schemas to my actions, unless the response schema is dynamic. 
- [ ] I validated the swagger file, `apiDefinition.swagger.json`, by running `paconn validate` command.
- [ ] If this is a certified connector, I confirm that `apiProperties.json` has a valid brand color and doesn't use an invalid brand color, `#007ee5` or `#ffffff`. If this is an independent publisher connector, I confirm that I am not submitting a connector icon.

If you are an Independent Publisher, you must also attest to the following to ensure a smooth publishing process:
- [ ] I have named this PR after the pattern of "Connector Name (Independent Publisher)" ex: HubSpot Marketing (Independent Publisher)
- [ ] Within this PR markdown file, I have pasted screenshots that show: 3 unique operations (actions/triggers) working within a Flow. This can be in one flow or part of multiple flows. For each one of those flows, I have pasted in screenshots of the Flow succeeding. 
- [ ] Within this PR markdown file, I have pasted in a screenshot from the Test operations section within the Custom Connector UI.
- [ ] If the connector uses OAuth, I have provided detailed steps on how to create an app in the readme.md.

jokano12 committed:
Update apiDefinition.swagger.json

lordlydon commented on certified-connectors/Impexium/apiDefinition.swagger.json:
```diff
@@ -16,6 +16,325 @@
+            ]
+        }
+    }
+},
```
There is an extra closing curly bracket "}" here

lordlydon commented on certified-connectors/Impexium/apiDefinition.swagger.json:
```diff
@@ -16,6 +16,325 @@
+                "200": {
+                    "description": "default",
+                    "schema": {
+                        "$ref": "#/definitions/EmailData"
```
The definition for EmailData is missing, and this endpoint returns an empty response.

lordlydon commented on certified-connectors/Impexium/apiDefinition.swagger.json:
```diff
@@ -16,6 +16,325 @@
+                    "in": "body",
+                    "required": true,
+                    "schema": {
+                        "$ref": "#/definitions/AddressSaveData"
```
The definition for AddressSaveData is missing

lordlydon commented on certified-connectors/Impexium/apiDefinition.swagger.json:
```diff
@@ -16,6 +16,325 @@
+                "200": {
+                    "description": "default",
+                    "schema": {
+                        "$ref": "#/definitions/AddressSaveData"
```
The definition for AddressSaveData is missing

lordlydon commented on certified-connectors/Impexium/apiDefinition.swagger.json:
```diff
@@ -16,6 +16,325 @@
+                            "dataList": {
+                                "type": "array",
+                                "items": {
+                                    "$ref": "#/definitions/CommitteeMemberDetailData"
```
The definition for CommitteeMemberDetailData is missing

lordlydon commented on certified-connectors/Impexium/apiDefinition.swagger.json:
```diff
@@ -16,6 +16,325 @@
   "consumes": ["application/json"],
   "produces": ["application/json"],
   "paths": {
+    "/api/v1/Organizations/{Page Number}": {
```
I'm getting an error of a duplicated path when adding this to the current connector.
It seems to be because it clashes with the `/api/v1/Organizations/{id}` route.
Instead of defining this here, you can add the get to that existing path definition below.

lordlydon replied:
After adding this to the other route, you need to remove this one, or the error will still occur.

jokano12 committed:
Update apiDefinition.swagger.json

Added Missing Refs, deleted curly brace , and moved the API call to under preexisting action.

lordlydon commented on certified-connectors/Impexium/apiDefinition.swagger.json:
```diff
@@ -2394,8 +2713,71 @@
+                }
+            ]
+        }
+    } },
```
There is an extra closing curly brace here

lordlydon commented on certified-connectors/Impexium/apiDefinition.swagger.json:
```diff
@@ -16,6 +16,325 @@
+                    "x-ms-summary": "Page Number"
+                },
+                {
+                    "name": "includeEmail",
```
When running paconn cli validate tool I'm getting this error for this property:
`The 'x-ms-summary' property is required.`

lordlydon commented on certified-connectors/Impexium/apiDefinition.swagger.json:
```diff
@@ -16,6 +16,325 @@
+                    "x-ms-summary": "Page Number"
+                },
+                {
+                    "name": "Term",
```
When running paconn cli validate tool I'm getting these errors for this property:
`The 'x-ms-summary' property is required.`
`The 'description' property is required.`

lordlydon commented on certified-connectors/Impexium/apiDefinition.swagger.json:
```diff
@@ -16,6 +16,325 @@
+                    "type": "integer"
+                },
+                {
+                    "name": "Position Codes",
```
When running paconn cli validate tool I'm getting these errors for this property:
`The 'x-ms-summary' property is required.`
`The 'description' property is required.`

lordlydon commented on certified-connectors/Impexium/apiDefinition.swagger.json:
```diff
@@ -2394,8 +2713,71 @@
-      }
-    },
+      },
+   "get": {
```
Can you fix the indentation here?

lordlydon commented on certified-connectors/Impexium/apiDefinition.swagger.json:
```diff
@@ -2394,8 +2713,71 @@
+                    "x-ms-summary": "Page Number"
+                },
+                {
+                    "name": "Name",
```
When running paconn cli validate tool I'm getting these errors for this property:
`The 'x-ms-summary' property is required.`

lordlydon commented on certified-connectors/Impexium/apiDefinition.swagger.json:
```diff
@@ -2394,8 +2713,71 @@
+                    "description": "Organization Name."
+                },
+                {
+                    "name": "Include Details",
```
When running paconn cli validate tool I'm getting these errors for this property:
`The 'x-ms-summary' property is required.`
`The 'description' property is required.`

lordlydon commented on certified-connectors/Impexium/apiDefinition.swagger.json:
```diff
@@ -2394,8 +2713,71 @@
+                    ]
+                },
+                {
+                    "name": "Tag",
```
When running paconn cli validate tool I'm getting these errors for this property:
`The 'x-ms-summary' property is required.`
`The 'description' property is required.`

lordlydon commented on certified-connectors/Impexium/apiDefinition.swagger.json:
```diff
@@ -2394,8 +2713,71 @@
+                    "type": "string"
+                },
+                {
+                    "name": "Domain",
```
When running paconn cli validate tool I'm getting these errors for this property:
`The 'x-ms-summary' property is required.`

lordlydon commented on certified-connectors/Impexium/apiDefinition.swagger.json:
```diff
@@ -16,6 +16,325 @@
+    "/api/v1/Committees/{ID Or Code}/Members/{Page Number}": {
+        "get": {
+            "responses": {
+                "default": {
```
When running paconn cli validate tool I'm getting this error for this property:
`The 'default' response should not have schema definition. Schemas should be defined on expected responses only.`

lordlydon commented on certified-connectors/Impexium/apiDefinition.swagger.json:
```diff
@@ -16,6 +16,325 @@
+            "operationId": "Add-Email-to-Individual",
+            "parameters": [
+                {
+                    "name": "ID Or Record Number",
```
When running paconn cli validate tool I'm getting this warning for this property:
`Encoding for path parameter 'ID Or Record Number' is missing, which might lead to invalid requests. Use \"x-ms-url-encoding\": \"single\" to ensure the value is URL encoded.`

lordlydon commented on certified-connectors/Impexium/apiDefinition.swagger.json:
```diff
@@ -16,6 +16,325 @@
+            "operationId": "Add-or-Update-Address-to-Individual",
+            "parameters": [
+                {
+                    "name": "ID Or Record Number",
```
When running paconn cli validate tool I'm getting this warning for this property:
`Encoding for path parameter 'ID Or Record Number' is missing, which might lead to invalid requests. Use \"x-ms-url-encoding\": \"single\" to ensure the value is URL encoded.`

lordlydon commented on certified-connectors/Impexium/apiDefinition.swagger.json:
```diff
@@ -16,6 +16,325 @@
+            "operationId": "Find-members-or-individuals-by-first-and-last-name",
+            "parameters": [
+                {
+                    "name": "First Name",
```
When running paconn cli validate tool I'm getting this warning for this property:
`Encoding for path parameter 'First Name' is missing, which might lead to invalid requests. Use \"x-ms-url-encoding\": \"single\" to ensure the value is URL encoded.`

lordlydon commented on certified-connectors/Impexium/apiDefinition.swagger.json:
```diff
@@ -16,6 +16,325 @@
+                    "x-ms-summary": "First Name"
+                },
+                {
+                    "name": "Last Name",
```
When running paconn cli validate tool I'm getting this warning for this property:
`Encoding for path parameter 'Last Name' is missing, which might lead to invalid requests. Use \"x-ms-url-encoding\": \"single\" to ensure the value is URL encoded.`

lordlydon commented on certified-connectors/Impexium/apiDefinition.swagger.json:
```diff
@@ -16,6 +16,325 @@
+                    "x-ms-summary": "Last Name"
+                },
+                {
+                    "name": "Page Number",
```
When running paconn cli validate tool I'm getting this warning for this property:
`Encoding for path parameter 'Page Number' is missing, which might lead to invalid requests. Use \"x-ms-url-encoding\": \"single\" to ensure the value is URL encoded.`

lordlydon commented on certified-connectors/Impexium/apiDefinition.swagger.json:
```diff
@@ -16,6 +16,325 @@
+            "operationId": "Get-Committee-Members-by-committee-ID-or-code",
+            "parameters": [
+                {
+                    "name": "ID Or Code",
```
When running paconn cli validate tool I'm getting this warning for this property:
`Encoding for path parameter 'ID Or Code' is missing, which might lead to invalid requests. Use \"x-ms-url-encoding\": \"single\" to ensure the value is URL encoded.`

lordlydon commented on certified-connectors/Impexium/apiDefinition.swagger.json:
```diff
@@ -16,6 +16,325 @@
+                    "x-ms-summary": "Committee ID or Code"
+                },
+                {
+                    "name": "Page Number",
```
When running paconn cli validate tool I'm getting this warning for this property:
`Encoding for path parameter 'Page Number' is missing, which might lead to invalid requests. Use \"x-ms-url-encoding\": \"single\" to ensure the value is URL encoded.`

lordlydon commented on certified-connectors/Impexium/apiDefinition.swagger.json:
```diff
@@ -2394,8 +2713,71 @@
+            "operationId": "List-All-Organizations",
+            "parameters": [
+                {
+                    "name": "Page Number",
```
When running paconn cli validate tool I'm getting this warning for this property:
`Encoding for path parameter 'id' is missing, which might lead to invalid requests. Use \"x-ms-url-encoding\": \"single\" to ensure the value is URL encoded."`

lordlydon reviewed (changes requested):
I added some more comments.
Please check the power automate connector cli tool, which provides a validate operation to check if there are errors or warnings with the connector swagger definition: https://github.com/Microsoft/PowerPlatformConnectors/tree/master/tools/paconn-cli#validate-a-swagger-json

jokano12 commented:
List All Individuals
Get List of Active Subscriptions for Individuals
Get Sub Committees
Get Positions by Committee
Get All Committees
